### PR TITLE
Update to 1.16.3 and LibDP

### LINKED ISSUE #50: crash with 1.16.2
**Describe the bug**
game crashes on startup with exit code -1

**To Reproduce**
1.  launch the game with cotton and fabric API in the mods folder
2. it should crash

**Expected behavior** *Optional*
should launch without issue

**Screenshots** *Optional*

**Version of the (please complete the following information):**
 - Cotton 1.0.7
 - Fabric API 0.9.2+build.206
 - Minecraft 1.16.2

**Additional context** *Optional*
here's a crash log, it mentions cotton cauldron and libCD in it. If it's set to secret by mistake just inform me https://gist.github.com/Nathan22211/7ed916b7f5cc0441d69c015c3e11ea4f

### PR DESCRIPTION
**Pull request by magneticflux-**
Closes #50 

~~Not done yet, but making progress!~~

I think it's done? It compiles and runs at least!

### CODE REVIEW

**Comment by BrekiTomasson** [comment]
Sorry to bother you guys, but any news on when this might be merged with master and released? Cotton is one of very few mods holding me back from being able to upgrade my server to 1.16.3 right now.

**Comment by LemmaEOF** [comment]
Lemme take a look now that it's been marked ready.

File: build.gradle [hunk 1]
```diff
@@ -35,17 +35,18 @@ dependencies {
...
     def modules = ["player-events", "logging", "config", "datapack", "commons", "cauldron"]
     modules.each { module ->
         modApi ("io.github.cottonmc.cotton:cotton-$module:+") { transitive = false }
         include "io.github.cottonmc.cotton:cotton-$module:+"
     }
-    modApi ("io.github.cottonmc:Jankson-Fabric:2.0.1+j1.2.0") { transitive = false }
-    modImplementation "io.github.cottonmc:LibCD:2.5.0+1.16.1"
-    include "io.github.cottonmc:Jankson-Fabric:2.0.1+j1.2.0"
+    modApi ("io.github.cottonmc:Jankson-Fabric:3.0.1+j1.2.0") { transitive = false }
+    modImplementation "io.github.cottonmc:LibCD:3.0.3+1.16.3"
```
Comment (LemmaEOF) [inline build.gradle:47]: I don't *think* we need the LibCD dep anymore? I'll take a look through.

File: modules/cotton-cauldron/src/main/resources/fabric.mod.json [hunk 2]
```diff
@@ -23,10 +23,11 @@
   "depends": {
     "fabricloader": ">=0.4.0",
     "cotton-commons": "*",
-    "libcd": ">=2.0.0"
+    "libcd": ">=3.0.0",
+    "libdp": ">=1.0.0"
```
Comment (LemmaEOF) [inline modules/cotton-cauldron/src/main/resources/fabric.mod.json:27]: I don't think this actually hard-deps on libcd *or* libdp anymore. I'll take a look through and see, and if it does then I'll fix it myself.

**Review by LemmaEOF — APPROVED** [review]

**Pull request merged**